src/Cli/func/Actions/AzureActions/LogStreamAction.cs

```diff
@@ -92,14 +93,24 @@ public override async Task RunAsync()
 
         public async Task OpenLiveStreamInBrowser(Site functionApp, IEnumerable<ArmSubscription> allSubscriptions)
         {
-            if (!functionApp.AzureAppSettings.ContainsKey(ApplicationInsightsIKeySetting))
+            string iKey;
+
+            // First, check for a connection string. If it's not available, default to using the Instrumentation Key.
+            if (functionApp.AzureAppSettings.TryGetValue(ApplicationInsightsConnectionString, out var connectionString))
+            {
+                iKey = Utilities.ExtractIKeyFromConnectionString(connectionString);
+            }
+            else if (functionApp.AzureAppSettings.TryGetValue(ApplicationInsightsIKeySetting, out var instrumentationKey))
+            {
+                iKey = instrumentationKey;
```

Comment:
Should we consider adding a warning log to inform customers that they should consider migrating from using INSTRUMENTATIONKEY to CONNECTIONSTRING?

Reply:
Added a warning.